### ISSUE TITLE
[homematic] NPE on startup

### ISSUE THREAD

**Issue by alexf2015** [openhab/openhab-addons#17372]
After upgrading from openhab 4.1.x to 4.2.1 homegear binding always crashes on startup of openhab:

```
2024-09-05 06:51:23.184 [INFO ] [org.openhab.core.Activator          ] - Starting openHAB 4.2.1 (Release Build)
2024-09-05 06:51:26.693 [INFO ] [.core.internal.i18n.I18nProviderImpl] - Time zone set to 'Europe/Berlin'.
2024-09-05 06:51:26.744 [INFO ] [.core.internal.i18n.I18nProviderImpl] - Location set to '51.45122561763559,6.283836364746094'.
2024-09-05 06:51:26.750 [INFO ] [.core.internal.i18n.I18nProviderImpl] - Locale set to 'de_DE'.
2024-09-05 06:52:09.126 [INFO ] [el.core.internal.ModelRepositoryImpl] - Loading model 'jeelink.items'
2024-09-05 06:52:11.888 [INFO ] [el.core.internal.ModelRepositoryImpl] - Loading model 'solaredge.items'
2024-09-05 06:52:12.662 [INFO ] [el.core.internal.ModelRepositoryImpl] - Loading model 'homegear.items'
2024-09-05 06:52:13.698 [INFO ] [el.core.internal.ModelRepositoryImpl] - Loading model 'default.items'
2024-09-05 06:52:15.853 [INFO ] [el.core.internal.ModelRepositoryImpl] - Loading model 'mapdb.persist'
2024-09-05 06:52:20.158 [INFO ] [el.core.internal.ModelRepositoryImpl] - Loading model 'jeelink.things'
2024-09-05 06:52:20.509 [INFO ] [el.core.internal.ModelRepositoryImpl] - Loading model 'homematic.things'
2024-09-05 06:52:20.696 [INFO ] [el.core.internal.ModelRepositoryImpl] - Loading model 'solaredge.things'
2024-09-05 06:52:50.144 [INFO ] [.core.model.lsp.internal.ModelServer] - Started Language Server Protocol (LSP) service on port 5007
2024-09-05 06:52:52.465 [INFO ] [el.core.internal.ModelRepositoryImpl] - Loading model 'default.rules'
2024-09-05 06:53:06.831 [INFO ] [el.core.internal.ModelRepositoryImpl] - Loading model 'boot.rules'
2024-09-05 06:53:21.331 [INFO ] [e.automation.internal.RuleEngineImpl] - Rule engine started.
2024-09-05 06:53:33.011 [WARN ] [mmon.WrappedScheduledExecutorService] - Scheduled runnable ended with an exception:
java.lang.NullPointerException: Cannot invoke "org.openhab.binding.homematic.internal.communicator.HomematicGateway.cancelLoadAllDeviceMetadata()" because the return value of "org.openhab.binding.homematic.internal.handler.HomematicBridgeHandler.getGateway()" is null
        at org.openhab.binding.homematic.internal.discovery.HomematicDeviceDiscoveryService.stopScan(HomematicDeviceDiscoveryService.java:112) ~[?:?]
        at org.openhab.binding.homematic.internal.handler.HomematicBridgeHandler.disposeInternal(HomematicBridgeHandler.java:185) ~[?:?]
        at org.openhab.binding.homematic.internal.handler.HomematicBridgeHandler.initializeInternal(HomematicBridgeHandler.java:134) ~[?:?]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539) ~[?:?]
        at java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[?:?]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304) ~[?:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) [?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) [?:?]
        at java.lang.Thread.run(Thread.java:840) [?:?]
2024-09-05 06:55:31.309 [WARN ] [core.thing.internal.ThingManagerImpl] - A thing handler factory claims to support 'homematic:HG-BC-SC-Rd-WM-2' for thing 'homematic:HG-BC-SC-Rd-WM-2:homegear:LEQ0004711' for more than 120s, but the thing type can't be found in the registry. This should be fixed in the binding.
2024-09-05 06:55:31.317 [WARN ] [core.thing.internal.ThingManagerImpl] - Could not normalize configuration for 'homematic:HG-BC-SC-Rd-WM-2:homegear:LEQ0004711' because the thing type was not found in registry.
2024-09-05 06:55:31.410 [WARN ] [core.thing.internal.ThingManagerImpl] - A thing handler factory claims to support 'homematic:HG-BC-SC-Rd-WM-2' for thing 'homematic:HG-BC-SC-Rd-WM-2:homegear:LKF0004711' for more than 120s, but the thing type can't be found in the registry. This should be fixed in the binding.
2024-09-05 06:55:31.413 [WARN ] [core.thing.internal.ThingManagerImpl] - Could not normalize configuration for 'homematic:HG-BC-SC-Rd-WM-2:homegear:LKF0004711' because the thing type was not found in registry.
2024-09-05 06:55:31.453 [WARN ] [core.thing.internal.ThingManagerImpl] - A thing handler factory claims to support 'homematic:HG-BC-SC-Rd-WM-2' for thing 'homematic:HG-BC-SC-Rd-WM-2:homegear:LKF0004712' for more than 120s, but the thing type can't be found in the registry. This should be fixed in the binding.
2024-09-05 06:55:31.461 [WARN ] [core.thing.internal.ThingManagerImpl] - Could not normalize configuration for 'homematic:HG-BC-SC-Rd-WM-2:homegear:LKF0004712' because the thing type was not found in registry.
2024-09-05 06:55:31.502 [WARN ] [core.thing.internal.ThingManagerImpl] - A thing handler factory claims to support 'homematic:HG-BC-SC-Rd-WM-2' for thing 'homematic:HG-BC-SC-Rd-WM-2:homegear:NEQ0004711' for more than 120s, but the thing type can't be found in the registry. This should be fixed in the binding.
...
...
```

The bridge then remains in offline status with message "connection refused". A workaround is to disable the bridge and enable it again. This results in further exceptions but at least the bridge and all connected devices (things) switch to online and work as expected.

```
2024-09-05 07:45:56.237 [ERROR] [nal.common.AbstractInvocationHandler] - An error occurred while calling method 'ThingHandler.dispose()' on 'org.openhab.binding.homematic.internal.handler.HomematicBridgeHandler@1aeb073': Cannot invoke "org.openhab.binding.homematic.internal.communicator.HomematicGateway.cancelLoadAllDeviceMetadata()" because the return value of "org.openhab.binding.homematic.internal.handler.HomematicBridgeHandler.getGateway()" is null
java.lang.NullPointerException: Cannot invoke "org.openhab.binding.homematic.internal.communicator.HomematicGateway.cancelLoadAllDeviceMetadata()" because the return value of "org.openhab.binding.homematic.internal.handler.HomematicBridgeHandler.getGateway()" is null
        at org.openhab.binding.homematic.internal.discovery.HomematicDeviceDiscoveryService.stopScan(HomematicDeviceDiscoveryService.java:112) ~[?:?]
        at org.openhab.binding.homematic.internal.handler.HomematicBridgeHandler.disposeInternal(HomematicBridgeHandler.java:185) ~[?:?]
        at org.openhab.binding.homematic.internal.handler.HomematicBridgeHandler.dispose(HomematicBridgeHandler.java:177) ~[?:?]
        at jdk.internal.reflect.GeneratedMethodAccessor74.invoke(Unknown Source) ~[?:?]
        at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
        at java.lang.reflect.Method.invoke(Method.java:569) ~[?:?]
        at org.openhab.core.internal.common.AbstractInvocationHandler.invokeDirect(AbstractInvocationHandler.java:147) [bundleFile:?]
        at org.openhab.core.internal.common.Invocation.call(Invocation.java:52) [bundleFile:?]
        at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) [?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) [?:?]
        at java.lang.Thread.run(Thread.java:840) [?:?]
2024-09-05 07:45:56.254 [ERROR] [core.thing.internal.ThingManagerImpl] - Exception occurred while disposing handler of thing 'homematic:bridge:homegear': Cannot invoke "org.openhab.binding.homematic.internal.communicator.HomematicGateway.cancelLoadAllDeviceMetadata()" because the return value of "org.openhab.binding.homematic.internal.handler.HomematicBridgeHandler.getGateway()" is null
java.lang.NullPointerException: Cannot invoke "org.openhab.binding.homematic.internal.communicator.HomematicGateway.cancelLoadAllDeviceMetadata()" because the return value of "org.openhab.binding.homematic.internal.handler.HomematicBridgeHandler.getGateway()" is null
        at org.openhab.binding.homematic.internal.discovery.HomematicDeviceDiscoveryService.stopScan(HomematicDeviceDiscoveryService.java:112) ~[?:?]
        at org.openhab.binding.homematic.internal.handler.HomematicBridgeHandler.disposeInternal(HomematicBridgeHandler.java:185) ~[?:?]
        at org.openhab.binding.homematic.internal.handler.HomematicBridgeHandler.dispose(HomematicBridgeHandler.java:177) ~[?:?]
        at jdk.internal.reflect.GeneratedMethodAccessor74.invoke(Unknown Source) ~[?:?]
        at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
        at java.lang.reflect.Method.invoke(Method.java:569) ~[?:?]
        at org.openhab.core.internal.common.AbstractInvocationHandler.invokeDirect(AbstractInvocationHandler.java:147) [bundleFile:?]
        at org.openhab.core.internal.common.Invocation.call(Invocation.java:52) [bundleFile:?]
        at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) [?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) [?:?]
        at java.lang.Thread.run(Thread.java:840) [?:?]
2024-09-05 07:46:01.634 [INFO ] [ternal.communicator.client.RpcClient] - Interface 'WIRED' on gateway 'homegear' not available, disabling support
2024-09-05 07:46:01.642 [INFO ] [ternal.communicator.client.RpcClient] - Interface 'HMIP' on gateway 'homegear' not available, disabling support
2024-09-05 07:46:01.649 [INFO ] [ternal.communicator.client.RpcClient] - Interface 'CUXD' on gateway 'homegear' not available, disabling support
2024-09-05 07:46:01.662 [INFO ] [ternal.communicator.client.RpcClient] - Interface 'GROUP' on gateway 'homegear' not available, disabling support
2024-09-05 07:46:01.788 [INFO ] [ommunicator.AbstractHomematicGateway] - HmGatewayInfo[id=HOMEGEAR,type=Homegear,firmware=0.7.51-3497,address=VBC8884544,rf=true,wired=false,hmip=false,cuxd=false,group=false]

```

**Comment by maniac103** [comment]
It's #17206. If the newly added check triggers, disposeInternal() is called without gateway being assigned.

**Comment by lsiepel** [comment]
> It's #17206. If the newly added check triggers, disposeInternal() is called without gateway being assigned. The check either needs to happen later, or disposeInternal must become aware of gateway potentially being null.

Thanks, title was homegear. Do you come up with a PR?

**Comment by soenkekueper** [comment]
I'm sorry for this - did not recognized this while testing.
I've added a comment on getGateway() and fixed the DiscoveryService - as

**Comment by jlaur** [comment]
> It's #17206. If the newly added check triggers, disposeInternal() is called without gateway being assigned. The check either needs to happen later, or disposeInternal must become aware of gateway potentially being null.

The issue is reported for 4.2.1, but that PR is part of 4.3 only?

**Comment by maniac103** [comment]
> The issue is reported for 4.2.1, but that PR is part of 4.3 only?

That's a good point ;-)
From the lack of a debug message I thought we're hitting [this](https://github.com/openhab/openhab-addons/blob/4.2.x/bundles/org.openhab.binding.homematic/src/main/java/org/openhab/binding/homematic/internal/handler/HomematicBridgeHandler.java#L138) line, but actually it's [this](https://github.com/openhab/openhab-addons/blob/4.2.x/bundles/org.openhab.binding.homematic/src/main/java/org/openhab/binding/homematic/internal/handler/HomematicBridgeHandler.java#L134) one. It looks like connecting to the gateway failed [here](https://github.com/openhab/openhab-addons/blob/4.2.x/bundles/org.openhab.binding.homematic/src/main/java/org/openhab/binding/homematic/internal/communicator/HomematicGatewayFactory.java#L51). #17372 should cover both scenarios though.

**Comment by jlaur** [comment]
> From the lack of a debug message I thought we're hitting [this](https://github.com/openhab/openhab-addons/blob/4.2.x/bundles/org.openhab.binding.homematic/src/main/java/org/openhab/binding/homematic/internal/handler/HomematicBridgeHandler.java#L138) line, but actually it's [this](https://github.com/openhab/openhab-addons/blob/4.2.x/bundles/org.openhab.binding.homematic/src/main/java/org/openhab/binding/homematic/internal/handler/HomematicBridgeHandler.java#L134) one. It looks like connecting to the gateway failed [here](https://github.com/openhab/openhab-addons/blob/4.2.x/bundles/org.openhab.binding.homematic/src/main/java/org/openhab/binding/homematic/internal/communicator/HomematicGatewayFactory.java#L51). #17390 should cover both scenarios though.

Thanks for verifying!

**Comment by prodigy** [comment]
Is this fix already released in 4.2.3? I still have this issue and cannot save authentication information.

Manually editing the configuration on the file system and restarting openhab does not resolve the issue.

```
07:47:17.746 [WARN ] [mmon.WrappedScheduledExecutorService] - Scheduled runnable ended with an exception:
java.lang.NullPointerException: Cannot invoke "org.openhab.binding.homematic.internal.communicator.HomematicGateway.cancelLoadAllDeviceMetadata()" because the return value of "org.openhab.binding.homematic.internal.handler.HomematicBridgeHandler.getGateway()" is null
        at org.openhab.binding.homematic.internal.discovery.HomematicDeviceDiscoveryService.stopScan(HomematicDeviceDiscoveryService.java:112) ~[?:?]
        at org.openhab.binding.homematic.internal.handler.HomematicBridgeHandler.disposeInternal(HomematicBridgeHandler.java:185) ~[?:?]
        at org.openhab.binding.homematic.internal.handler.HomematicBridgeHandler.initializeInternal(HomematicBridgeHandler.java:134) ~[?:?]
07:47:45.506 [ERROR] [st.core.internal.thing.ThingResource] - Exception during HTTP PUT request for update config at 'things/homematic:bridge:3014F711A0001F9F29B2C5F2/config'
java.lang.NullPointerException: Cannot invoke "org.openhab.binding.homematic.internal.communicator.HomematicGateway.cancelLoadAllDeviceMetadata()" because the return value of "org.openhab.binding.homematic.internal.handler.HomematicBridgeHandler.getGateway()" is null
        at org.openhab.binding.homematic.internal.discovery.HomematicDeviceDiscoveryService.stopScan(HomematicDeviceDiscoveryService.java:112) ~[?:?]
        at org.openhab.binding.homematic.internal.handler.HomematicBridgeHandler.disposeInternal(HomematicBridgeHandler.java:185) ~[?:?]
        at org.openhab.binding.homematic.internal.handler.HomematicBridgeHandler.dispose(HomematicBridgeHandler.java:177) ~[?:?]
        at org.openhab.core.thing.binding.BaseThingHandler.handleConfigurationUpdate(BaseThingHandler.java:110) ~[?:?]
        at org.openhab.core.thing.internal.ThingRegistryImpl.updateConfiguration(ThingRegistryImpl.java:94) ~[?:?]
        at org.openhab.core.io.rest.core.internal.thing.ThingResource.updateConfiguration(ThingResource.java:538) ~[?:?]
        at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
        at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[?:?]
        at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
        at java.lang.reflect.Method.invoke(Method.java:569) ~[?:?]
        at org.apache.cxf.service.invoker.AbstractInvoker.performInvocation(AbstractInvoker.java:179) ~[bundleFile:3.6.2]
        at org.apache.cxf.service.invoker.AbstractInvoker.invoke(AbstractInvoker.java:96) ~[bundleFile:3.6.2]
        at org.apache.cxf.jaxrs.JAXRSInvoker.invoke(JAXRSInvoker.java:201) ~[bundleFile:3.6.2]
        at org.apache.cxf.jaxrs.JAXRSInvoker.invoke(JAXRSInvoker.java:104) ~[bundleFile:3.6.2]
        at org.apache.cxf.interceptor.ServiceInvokerInterceptor$1.run(ServiceInvokerInterceptor.java:59) ~[bundleFile:3.6.2]
        at org.apache.cxf.interceptor.ServiceInvokerInterceptor.handleMessage(ServiceInvokerInterceptor.java:96) ~[bundleFile:3.6.2]
        at org.apache.cxf.phase.PhaseInterceptorChain.doIntercept(PhaseInterceptorChain.java:307) ~[bundleFile:3.6.2]
        at org.apache.cxf.transport.ChainInitiationObserver.onMessage(ChainInitiationObserver.java:121) ~[bundleFile:3.6.2]
        at org.apache.cxf.transport.http.AbstractHTTPDestination.invoke(AbstractHTTPDestination.java:265) ~[bundleFile:3.6.2]
        at org.apache.cxf.transport.servlet.ServletController.invokeDestination(ServletController.java:234) ~[bundleFile:3.6.2]
        at org.apache.cxf.transport.servlet.ServletController.invoke(ServletController.java:208) ~[bundleFile:3.6.2]
        at org.apache.cxf.transport.servlet.ServletController.invoke(ServletController.java:160) ~[bundleFile:3.6.2]
        at org.apache.cxf.transport.servlet.CXFNonSpringServlet.invoke(CXFNonSpringServlet.java:225) ~[bundleFile:3.6.2]
        at org.apache.cxf.transport.servlet.AbstractHTTPServlet.handleRequest(AbstractHTTPServlet.java:304) ~[bundleFile:3.6.2]
        at org.apache.cxf.transport.servlet.AbstractHTTPServlet.doPut(AbstractHTTPServlet.java:234) ~[bundleFile:3.6.2]
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:520) ~[bundleFile:4.0.4]
        at org.apache.cxf.transport.servlet.AbstractHTTPServlet.service(AbstractHTTPServlet.java:279) ~[bundleFile:3.6.2]
        at org.ops4j.pax.web.service.spi.servlet.OsgiInitializedServlet.service(OsgiInitializedServlet.java:102) ~[bundleFile:?]
        at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:799) ~[bundleFile:9.4.54.v20240208]
        at org.eclipse.jetty.servlet.ServletHandler$ChainEnd.doFilter(ServletHandler.java:1656) ~[bundleFile:9.4.54.v20240208]
        at org.ops4j.pax.web.service.spi.servlet.OsgiFilterChain.doFilter(OsgiFilterChain.java:113) ~[bundleFile:?]
        at org.ops4j.pax.web.service.jetty.internal.PaxWebServletHandler.doHandle(PaxWebServletHandler.java:334) ~[bundleFile:?]
        at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143) ~[bundleFile:9.4.54.v20240208]
        at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:600) ~[bundleFile:9.4.54.v20240208]
        at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127) ~[bundleFile:9.4.54.v20240208]
        at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:235) ~[bundleFile:9.4.54.v20240208]
        at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1624) ~[bundleFile:9.4.54.v20240208]
        at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:233) ~[bundleFile:9.4.54.v20240208]
        at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1440) ~[bundleFile:9.4.54.v20240208]
        at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:188) ~[bundleFile:9.4.54.v20240208]
        at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:505) ~[bundleFile:9.4.54.v20240208]
        at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1594) ~[bundleFile:9.4.54.v20240208]
        at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:186) ~[bundleFile:9.4.54.v20240208]
        at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1355) ~[bundleFile:9.4.54.v20240208]
        at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141) ~[bundleFile:9.4.54.v20240208]
        at org.eclipse.jetty.server.handler.ContextHandlerCollection.handle(ContextHandlerCollection.java:234) ~[bundleFile:9.4.54.v20240208]
        at org.ops4j.pax.web.service.jetty.internal.PrioritizedHandlerCollection.handle(PrioritizedHandlerCollection.java:96) ~[bundleFile:?]
        at org.eclipse.jetty.server.handler.gzip.GzipHandler.handle(GzipHandler.java:722) ~[bundleFile:9.4.54.v20240208]
        at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127) ~[bundleFile:9.4.54.v20240208]
        at org.eclipse.jetty.server.Server.handle(Server.java:516) ~[bundleFile:9.4.54.v20240208]
        at org.eclipse.jetty.server.HttpChannel.lambda$handle$1(HttpChannel.java:487) ~[bundleFile:9.4.54.v20240208]
        at org.eclipse.jetty.server.HttpChannel.dispatch(HttpChannel.java:732) [bundleFile:9.4.54.v20240208]
        at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:479) [bundleFile:9.4.54.v20240208]
        at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:277) [bundleFile:9.4.54.v20240208]
        at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:311) [bundleFile:9.4.54.v20240208]
        at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:105) [bundleFile:9.4.54.v20240208]
        at org.eclipse.jetty.io.ssl.SslConnection$DecryptedEndPoint.onFillable(SslConnection.java:555) [bundleFile:9.4.54.v20240208]
        at org.eclipse.jetty.io.ssl.SslConnection.onFillable(SslConnection.java:410) [bundleFile:9.4.54.v20240208]
        at org.eclipse.jetty.io.ssl.SslConnection$2.succeeded(SslConnection.java:164) [bundleFile:9.4.54.v20240208]
        at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:105) [bundleFile:9.4.54.v20240208]
        at org.eclipse.jetty.io.ChannelEndPoint$1.run(ChannelEndPoint.java:104) [bundleFile:9.4.54.v20240208]
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:338) [bundleFile:9.4.54.v20240208]
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:315) [bundleFile:9.4.54.v20240208]
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:173) [bundleFile:9.4.54.v20240208]
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:131) [bundleFile:9.4.54.v20240208]
        at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:409) [bundleFile:9.4.54.v20240208]
```
